build-profiling-ffi.sh

```diff
@@ -169,6 +169,25 @@ if [[ "$fix_macos_rpath" -eq 1 ]]; then
     install_name_tool -id @rpath/${shared_library_name} "$destdir/lib/${shared_library_name}"
 fi
 
+if command -v patchelf > /dev/null && [[ "$target" != "x86_64-pc-windows-msvc" ]]; then
+    patchelf_bin=`which patchelf`
+    # This is temporary (until we have the new builder everywhere) and can avoid using patchelf
+    # On aarch64 centos specifically, patchelf is old.
+    # And when applying strip after patchelf, strip fails because the binary looks corrupted
+    # In this specific case, we download the latest version of patchelf for aarch64 and use it
+    # https://github.com/NixOS/patchelf/pull/117
+    if [[ "$target" == "aarch64-unknown-linux-gnu" ]]; then
+        mkdir patchelf
+        cd patchelf
+        curl -OL https://github.com/NixOS/patchelf/releases/download/0.18.0/patchelf-0.18.0-aarch64.tar.gz
+        tar zxvf patchelf-0.18.0-aarch64.tar.gz
+        patchelf_bin=`pwd`"/bin/patchelf"
+        cd ..
+    fi
```

Comment:
Minor: Maybe add here a `echo "Downloading patchelf to workaround bug in centos patchelf"` in case someone gets surprised by this?

Reply:
I updated the Docker image instead so we only have the call to `patchelf`